lib/mix/lib/mix/scm/git.ex

```diff
@@ -12,6 +12,7 @@ defmodule Mix.SCM.Git do
     else
       opts[:git]
     end
+    |> redact_git_url
```

Comment:
Instead of parsing the whole text with a regex, my suggestion is to use the functions in the URI module, something like this:

```elixir
defp redact_uri(git) do
  case URI.parse(git) do
    %{username: nil, password: nil} -> git
    uri -> URI.to_string(%{uri | username: "****", password: "****"})
  end
end
```

Then you call it with `opts[:git]` on both `format` and `format_lock`. WDYT?